test/integration/test_messages_api.py

```diff
@@ -0,0 +1,187 @@
+            to=['+invalid'],
+            _from='+invalid',
+            text='',
+        )
+        
+        with self.assertRaises(UnauthorizedException) as context:
+            api_instance.create_message(account_id, message_request)
+             
+        self.assertEqual(context.exception.status, 401)
+
+        e = MessagingRequestError(**json.loads(context.exception.body))
+        self.assertIsInstance(e, MessagingRequestError)
+        self.assertEqual(e.type, 'unauthorized')
+        self.assertEqual(e.description, 'Authentication Failed')
+
+    # The SDK catches incorrect content-type before making the request and attempts to create an ApiException,
+    # but the creation of the exception fails since there is no response body. This should probably create some
+    # kind of Client Exception instead, since this isn't an actual API Exception.
+
+    def test_list_messages(self):
+        account_id = os.environ.get('BW_ACCOUNT_ID')
+        message_direction = ListMessageDirectionEnum("OUTBOUND")
+        
+        try:
+            api_response = self.api_instance.list_messages(account_id, message_direction=message_direction)
+        except bandwidth.ApiException as e:
+            print("Exception when calling MessagesApi->list_messages: %s\n" % e)
+
+        self.assertIsInstance(api_response, MessagesList)
+        self.assertGreater(api_response.total_count, 0)
+        self.assertTrue(api_response.messages[0], ListMessageItem)
+        
+        message = api_response.messages[0]
+        self.assertEqual(message.account_id, os.environ.get('BW_ACCOUNT_ID'))
+        self.assertTrue(message.carrier_name)
+        self.assertRegex(message.destination_tn, '^\\+[1-9]\\d{1,14}$')
+        self.assertFalse(message.error_code)
+        self.assertIsInstance(message.message_direction, ListMessageDirectionEnum)
+        self.assertTrue(message.message_id)
+        self.assertIsInstance(message.message_status, MessageStatusEnum)
+        self.assertIsInstance(message.message_type, MessageTypeEnum)
+        self.assertTrue(datetime.fromisoformat(message.receive_time[:-1]))
+        self.assertTrue(message.segment_count)
+        self.assertRegex(message.source_tn, '^\\+[1-9]\\d{1,14}$')
+
+    
+    def test_list_messages_bad_request(self):
+        account_id = os.environ.get('BW_ACCOUNT_ID')
+        
+        with self.assertRaises(ApiException) as context:
+            self.api_instance.list_messages(account_id)
+        
+        self.assertEqual(context.exception.status, 400)
+
+        e = MessagingRequestError(**json.loads(context.exception.body))
+        self.assertIsInstance(e, MessagingRequestError)
+        self.assertEqual(e.type, 'bad-request')
+        self.assertIsInstance(e.description, str)
+
+    
+    def test_list_messages_unauthorized(self):
+        api_client = bandwidth.ApiClient()
+        api_instance = messages_api.MessagesApi(api_client)
```

Comment:
```suggestion
        unauthorized_api_client = bandwidth.ApiClient()
        api_instance = messages_api.MessagesApi(unauthorized_api_client)
```
just to be more explicit